crates/router/src/core/payment_methods/vault.rs

```diff
@@ -1151,133 +1271,12 @@ pub async fn delete_tokenized_data(
-    )
-    .into_report()
-    .change_context(errors::ApiErrorResponse::InternalServerError)
-    .attach_printable_lazy(|| {
-        format!(
-            "unable to convert into DeleteTokenizeByTokenRequest {:?}",
-            tokenize_tracker.tracking_data
-        )
-    })?;
-
-    let delete_resp = delete_tokenized_data(state, &delete_tokenize_data.lookup_key).await;
-    match delete_resp {
-        Ok(resp) => {
-            if resp == "Ok" {
-                logger::info!("Card From locker deleted Successfully");
-                //mark task as finished
-                let id = tokenize_tracker.id.clone();
-                tokenize_tracker
-                    .clone()
-                    .finish_with_status(db.as_scheduler(), format!("COMPLETED_BY_PT_{id}"))
-                    .await?;
-            } else {
-                logger::error!("Error: Deleting Card From Locker : {:?}", resp);
-                retry_delete_tokenize(db, &delete_tokenize_data.pm, tokenize_tracker.to_owned())
-                    .await?;
-                metrics::RETRIED_DELETE_DATA_COUNT.add(&metrics::CONTEXT, 1, &[]);
-            }
-        }
-        Err(err) => {
-            logger::error!("Err: Deleting Card From Locker : {:?}", err);
-            retry_delete_tokenize(db, &delete_tokenize_data.pm, tokenize_tracker.to_owned())
-                .await?;
-            metrics::RETRIED_DELETE_DATA_COUNT.add(&metrics::CONTEXT, 1, &[]);
-        }
-    }
-    Ok(())
-}
-
 #[cfg(feature = "basilisk")]
-pub async fn get_delete_tokenize_schedule_time(
-    db: &dyn db::StorageInterface,
-    pm: &enums::PaymentMethod,
-    retry_count: i32,
-) -> Option<time::PrimitiveDateTime> {
-    let redis_mapping = db::get_and_deserialize_key(
-        db,
-        &format!("pt_mapping_delete_{pm}_tokenize_data"),
-        "PaymentMethodsPTMapping",
-    )
-    .await;
-    let mapping = match redis_mapping {
-        Ok(x) => x,
-        Err(err) => {
-            logger::info!("Redis Mapping Error: {}", err);
-            process_data::PaymentMethodsPTMapping::default()
-        }
-    };
-    let time_delta = process_tracker_utils::get_pm_schedule_time(mapping, pm, retry_count + 1);
-
-    process_tracker_utils::get_time_from_delta(time_delta)
-}
-
-#[cfg(feature = "basilisk")]
-pub async fn retry_delete_tokenize(
-    db: &dyn db::StorageInterface,
-    pm: &enums::PaymentMethod,
-    pt: storage::ProcessTracker,
-) -> Result<(), errors::ProcessTrackerError> {
-    let schedule_time = get_delete_tokenize_schedule_time(db, pm, pt.retry_count).await;
-
-    match schedule_time {
-        Some(s_time) => pt.retry(db.as_scheduler(), s_time).await,
-        None => {
-            pt.finish_with_status(db.as_scheduler(), "RETRIES_EXCEEDED".to_string())
-                .await
-        }
+pub fn get_key_id(keys: &settings::Jwekey) -> &str {
+    let key_identifier = "1"; // [#46]: Fetch this value from redis or external sources
```

Comment:
Why is this hardcoded?

Reply:
No idea. Picked it up from old basilisk code. After temp locker works successfully , we will remove it anyways.https://github.com/juspay/hyperswitch/pull/2789/files/56acdcb4f96e5d36ce6b87f8ceb10e3ba30e48de#diff-1fb209f2e84bcd1f319046b6b49cb5d3b63a972d1272147823c4ff5e2731ab34L912-L917